includes/admin/class-wcs-admin-meta-boxes.php

```diff
@@ -85,8 +87,12 @@ public function add_meta_boxes() {
 		add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_subscription', 'normal', 'low' );
 
 		// Only display the meta box if an order relates to a subscription
-		if ( 'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID ) && wcs_order_contains_subscription( $post_ID, 'any' ) ) {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			// If HPOS is enabled, we need to get the screen id and provide it to add_meta_box().
+			$screen = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled()
```

Comment:
We have a helper function for this now: `wcs_is_custom_order_tables_usage_enabled()`

Reply:
Nice! I knew that line would break spectacularly when I tested backwards-compat but didn't want to wrap it in yet another if/ternary 😂